File: server/src/main/java/com/example/backend/modules/quiz/models/Question.java
package com.example.backend.modules.quiz.models;


import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;
import java.util.List;

@Builder
@NoArgsConstructor
@AllArgsConstructor
@Data
@Entity
@Table(name = "questions")
public class Question {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;


    @Column(length = 150,nullable = false)
    private String title;

    @Column(columnDefinition = "MEDIUMTEXT",nullable = true)
    private String thumbnail;

    @Column(nullable = false)
    private int score;

    @Column(nullable = false,name = "order_number")
    private int order;

    @Column(nullable = false,name = "time_limit_seconds")
    private int timeLimit;

    @Column(nullable = false,name = "created_at")
    private Date createdAt;

    @Column(nullable = false,name = "updated_at")
    private Date updatedAt;

    @ManyToOne
    @JoinColumn(name = "quiz_id",nullable = false)
    @JsonBackReference
    private Quiz quiz;

    @ManyToOne
    @JoinColumn(name = "question_type_id",nullable = false)
    @JsonBackReference
    private QuestionType questionType;

    @OneToMany(fetch = FetchType.LAZY,cascade = CascadeType.ALL,mappedBy = "question")
    @JsonManagedReference
    private List<Answer> answers;
}

File: client/app/src/main/java/com/example/client/model/Question.kt
package com.example.client.model

import android.net.Uri

data class CreateQuestion (
    val title:String,
    val questionType:QuestionType,
    val order:Int,
    val score:Int,
    val timeLimit:Int,
    val answers:List<CreateAnswer>,
    val thumbnail: String?,
    val uri: Uri?
)

data class QuestionDetail(
    val id:Int,
    val title:String,
    val score:Int,
    val timeLimit:Int,
    val answers:List<Answer>,
    val thumbnail: String?,
    val order:Int,
    val questionType:QuestionType,
)

data class Question(
    val id:Int,
    val title:String,
    val answers:List<Answer>,
    val thumbnail:String?,
    val score:Int
)

data class EditQuestion(
    val questionId:Int,
    val title:String,
    val score:Int,
    val timeLimit:Int,
    val quizId:Int,
    val questionTypeId:Int
)

data class EditQuestionThumbnail(
    val questionId:Int,
    val quizId:Int,
    val thumbnail:String
)

data class QuestionType(
    val id:Int,
    val title:String
)